tests/sentry/api/test_id_or_slug_path_parameters.py

```diff
@@ -366,6 +493,218 @@ def group_test(self, endpoint_class, slug_params, *args):
             endpoint_class, converted_slugs, converted_ids, reverse_non_slug_mappings
         )
 
+    @patch("sentry.api.bases.organization.OrganizationEndpoint.check_object_permissions")
+    @override_options({"api.id-or-slug-enabled": True})
+    def external_user_details_test(self, endpoint_class, slug_params, *args):
+        slug_kwargs = {param: self.slug_mappings[param].slug for param in slug_params}
+        id_kwargs = {param: self.slug_mappings[param].id for param in slug_params}
+
+        request = Request(request=self.make_request())
+
+        external_user = self.create_external_user(organization=self.organization)
+
+        non_slug_mappings: dict[str, Any] = {
+            "external_user_id": external_user.id,
+        }
+
+        reverse_non_slug_mappings: dict[str, Any] = {
+            "external_user": external_user,
+        }
+
+        _, converted_slugs = endpoint_class().convert_args(
+            request=request, **slug_kwargs, **non_slug_mappings
+        )
+        _, converted_ids = endpoint_class().convert_args(
+            request=request, **id_kwargs, **non_slug_mappings
+        )
+
+        self.assert_conversion(
+            endpoint_class, converted_slugs, converted_ids, reverse_non_slug_mappings
+        )
+
+    @patch("sentry.api.bases.organization.OrganizationEndpoint.check_object_permissions")
+    @patch(
+        "sentry.api.endpoints.organization_dashboard_details.OrganizationDashboardBase._get_dashboard"
+    )
+    @override_options({"api.id-or-slug-enabled": True})
+    def organization_dashboard_details_test(self, endpoint_class, slug_params, *args):
+        slug_kwargs = {param: self.slug_mappings[param].slug for param in slug_params}
+        id_kwargs = {param: self.slug_mappings[param].id for param in slug_params}
+
+        request = Request(request=self.make_request())
+
+        non_slug_mappings: dict[str, Any] = {
+            "dashboard_id": "dashboard.id",
+        }
+
+        _, converted_slugs = endpoint_class().convert_args(
+            request=request, **slug_kwargs, **non_slug_mappings
+        )
+        _, converted_ids = endpoint_class().convert_args(
+            request=request, **id_kwargs, **non_slug_mappings
+        )
+
+        converted_slugs.pop("dashboard")
+        converted_ids.pop("dashboard")
+
+        self.assert_conversion(endpoint_class, converted_slugs, converted_ids)
+
+    @patch("sentry.api.bases.organization.OrganizationEndpoint.check_object_permissions")
+    @patch("sentry.api.bases.organization.subdomain_is_region")
+    @with_feature("organizations:incidents")
+    @override_options({"api.id-or-slug-enabled": True})
+    def incident_test(self, endpoint_class, slug_params, *args):
```

Comment:
Is this named after the incident endpoint? We should try to match the names so it's immediately obvious which endpoint the test applies to